src/server/game/Battlegrounds/Battleground.cpp

```diff
@@ -916,9 +916,52 @@ void Battleground::EndBattleground(PvPTeamId winnerTeamId)
         player->UpdateAchievementCriteria(ACHIEVEMENT_CRITERIA_TYPE_COMPLETE_BATTLEGROUND, player->GetMapId());
     }
 
+    if (IsEventActive(EVENT_SPIRIT_OF_COMPETITION) && isBattleground())
+        SpiritofCompetitionEvent(winnerTeamId);
+
     sScriptMgr->OnBattlegroundEnd(this, GetTeamId(winnerTeamId));
 }
 
+bool Battleground::SpiritofCompetitionEvent(PvPTeamId winnerTeamId)
+{
+    // Everyone is eligeble for tabard reward
+    for (BattlegroundPlayerMap::const_iterator itr = m_Players.begin(); itr != m_Players.end(); ++itr)
+    {
+        Player* player = itr->second;
+        bool questStatus = player->GetQuestStatus(QUEST_FLAG_PARTICIPANT) != QUEST_STATUS_REWARDED;
+        if (player && questStatus)
+            player->CastSpell(player, SPELL_SPIRIT_OF_COMPETITION_PARTICIPANT, true);
+    }
+
+    // Incase of draw nobody get reward
+    if (winnerTeamId == PVP_TEAM_NEUTRAL)
+        return false;
+    else
+    {
+        std::vector<Player*> filteredPlayers;
+
+        for (BattlegroundPlayerMap::const_iterator itr = m_Players.begin(); itr != m_Players.end(); ++itr)
+        {
+            Player* player = itr->second;
+            bool playerTeam = player->GetBgTeamId() == GetTeamId(winnerTeamId);
+            bool questStatus = player->GetQuestStatus(QUEST_FLAG_WINNER) != QUEST_STATUS_REWARDED;
+            if (player && playerTeam && questStatus)
+            {
+                filteredPlayers.push_back(player);
+            }
+        }
+
+        if (filteredPlayers.size())
+        {
+            Player* wPlayer = filteredPlayers[rand() % filteredPlayers.size()];
+
+            wPlayer->CastSpell(wPlayer, SPELL_SPIRIT_OF_COMPETITION_WINNER, true);
+        }
+    }
+
```

Comment:
Just a small nitpick.

```suggestion

    std::vector<Player*> filteredPlayers;
    
    for (BattlegroundPlayerMap::const_iterator itr = m_Players.begin(); itr != m_Players.end(); ++itr)
    {
        Player* player = itr->second;
        bool playerTeam = player->GetBgTeamId() == GetTeamId(winnerTeamId);
        bool questStatus = player->GetQuestStatus(QUEST_FLAG_WINNER) != QUEST_STATUS_REWARDED;
        if (player && playerTeam && questStatus)
        {
            filteredPlayers.push_back(player);
        }
    }
    
    if (filteredPlayers.size())
    {
        Player* wPlayer = filteredPlayers[rand() % filteredPlayers.size()];
    
        wPlayer->CastSpell(wPlayer, SPELL_SPIRIT_OF_COMPETITION_WINNER, true);
    }
```

Reply:
AH that's fair